src/hsbalance/CI_matrix.py

```diff
@@ -78,7 +78,7 @@ def add(self, direct_matrix:'np.array'=None, A:'initial_vibration numpy.array'=N
                     self.B = B
                     self.U = U
                     if not keep_trial:
-                        self.value = (self.B - self.A) / self.U
+                        self.value = (self.B - self.A) @ np.linalg.inv(self.U)
```

Comment:
Great work @shoaibmahmod7 
Inverse matrix will only work on square matrices. I think to do your implementation of multiple trial weights on more than one  plane, we need to make the square matrix to be the default.
So the original trial vector can be a square zero matrix with the diagonal to be the values of the trials at each plane.
U = [1, 2, 3]  --> U =[[1, 0, 0], [0, 2, 0], [0, 0, 3]]
This will not break the structure of the package and still we can apply your idea.
I think this is doable.